### PR TITLE
Added support for custom Directors

### PR DESCRIPTION
**Pull request by hynd**
Hello!
Custom director's aren't particularly well documented on Fastly's site yet, but essential for anything other than per-request random balancing across origins.

### CODE REVIEW

File: website/docs/r/service_v1.html.markdown [hunk 1]
```diff
@@ -119,6 +117,40 @@ resource "fastly_service_v1" "demo" {
 }
 ```
 
+Basic usage with [custom Director](https://docs.fastly.com/api/config#director):
```
Comment (thommahoney) [inline website/docs/r/service_v1.html.markdown:122]: Thanks for your thorough docs. I may not hold onto this section specifically long term. I think we can document each of the API resources without providing full examples for each of them.

**Review by thommahoney — DISMISSED** [review]

**Comment by mgue** [comment]
Hi @thommahoney any idea when we can get this merged?

**Comment by thommahoney** [comment]
Sorry for the delayed response. Hoping to have an answer for you soon.

**Comment by hynd** [comment]
Hi @thommahoney - any progress on an answer for this?  Thanks!

**Comment by lanej** [comment]
@hynd if you can resolve the conflicts, we can re-review this PR.  Thanks for your patience.

**Comment by hynd** [comment]
Sure can! Rebased...

File: fastly/resource_fastly_service_v1.go [hunk 2]
```diff
@@ -1350,7 +1405,7 @@ func resourceServiceV1Update(d *schema.ResourceData, meta interface{}) error {
 					Name:    cf["name"].(string),
 				}
 
-				log.Printf("[DEBUG] Fastly Conditions Removal opts: %#v", opts)
+				log.Printf("[DEBUG] Condition Removal opts: %#v", opts)
```
Comment (lanej) [inline fastly/resource_fastly_service_v1.go]: While I do agree that adding 'Fastly' to the log line is redundant, it has no relation to the intended change and should be done in another PR with a dedicated purpose.

The same is true for the remaining changes below.

File: website/docs/r/service_v1.html.markdown [hunk 3]
```diff
@@ -217,6 +250,15 @@ used in the `request_condition`, `response_condition`, or
 * `priority` - (Optional) A number used to determine the order in which multiple
 conditions execute. Lower numbers execute first. Default `10`.
 
+The `director` block supports:
```
Comment (JakeChampion) [inline website/docs/r/service_v1.html.markdown:255]: Should the director block support all the fields that the director api has available? https://docs.fastly.com/api/config#director

**Review by lanej — DISMISSED** [review]

File: website/docs/r/service_v1.html.markdown (hunk 3 above)
Comment (lanej) [reply website/docs/r/service_v1.html.markdown:255]: That would be ideal but not required (cc/ @hynd).  I would prefer to prioritize the relief of the users for features that can easily be augmented in the future.

Comment (hynd) [reply website/docs/r/service_v1.html.markdown:255]: Good spot - `comment` was missing from the doc, i *think* the only thing actually missing from the code is the `shield` field (which isn't in the underlying go-fastly lib.... yet)

Comment (hynd) [reply website/docs/r/service_v1.html.markdown:255]: Ah, and `capacity` is only half-implemented in go-fastly - i'll PR fixes for both of those first, and will update this accordingly....

Comment (lanej) [reply website/docs/r/service_v1.html.markdown:255]: Thanks @hynd

Comment (hynd) [reply website/docs/r/service_v1.html.markdown:255]: Done - merged upstream and refreshed this PR

**Comment by lanej** [comment]
Got an acceptance test failure:

```
--- FAIL: TestAccFastlyServiceV1_directors_basic (18.96s)
	testing.go:434: Step 0 error: Check failed: Check 2/4 error: Bad match Director match, expected (&fastly.Director{ServiceID:"3UIdDTdTSwAh1AZ1rw20RS", Version:1, Name:"mydirector", Comment:"", Shield:"", Quorum:0x0, Type:0x3, Retries:0x0, Capacity:0x0, DeletedAt:"", CreatedAt:"", UpdatedAt:""}), got (&fastly.Director{ServiceID:"3UIdDTdTSwAh1AZ1rw20RS", Version:1, Name:"mydirector", Comment:"", Shield:"", Quorum:0x4b, Type:0x3, Retries:0x5, Capacity:0x64, DeletedAt:"", CreatedAt:"2018-09-19T17:35:52Z", UpdatedAt:"2018-09-19T17:35:52Z"})
```

Looks like the timestamps don't match.  I haven't dug into the reason but if you could investigate, I would appreciate it.

**Comment by hynd** [comment]
Fixed!  Also removed the type "2" (round-robin) director, which seems to have been recently dropped from the docs (and throws an HTTP 400).

**Review by lanej — APPROVED** [review]

**Comment by lanej** [comment]
Thanks @hynd
